src/components/Inventory/InventoryDetails.js

```diff
@@ -0,0 +1,124 @@
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+
+import BackIcon from "../../assets/icons/arrow_back-24px.svg";
+import EditIcon from "../../assets/icons/edit-24px.svg";
+
+import "./InventoryDetails.scss";
+
+const InventoryDetails = () => {
+  const { id } = useParams();
+  const [inventoryDetails, setInventoryDetails] = useState({});
+  const [detailsLoaded, setDetailsLoaded] = useState(false);
+  const navigate = useNavigate();
+
+  const statusClass = inventoryDetails.status
+    ? inventoryDetails.status.split(" ").join("-").toLowerCase()
+    : ""; // Safely compute statusClass only after ensuring the data exists
+
+  // This useEffect fetches the inventory data when the component mounts
+  useEffect(() => {
+    const fetchInventoryDetails = async () => {
+      try {
+        const resp = await axios.get(
+          `http://localhost:8080/api/inventories/${id}`
+        );
+        setInventoryDetails(resp.data);
+        setDetailsLoaded(true);
+      } catch (error) {
+        console.error("Error fetching inventory data:", error);
+      }
+    };
+    fetchInventoryDetails();
+  });
```

Comment:
You are missing dependency array here. And thats why the network call to this page is infinite and its almost crashing my page.
Always have network tab open to ensure these things are not missed.

 ```
fetchInventoryDetails();
  }, []);
```